controller/aks-cluster-config-handler.go

```diff
@@ -87,35 +84,26 @@ func (h *Handler) OnAksConfigRemoved(key string, config *aksv1.AKSClusterConfig)
 	ctx := context.Background()
 	logrus.Infof("Removing cluster [%s]", config.Spec.ClusterName)
 
-	azureAuthorizer, err := newClientAuthorizer(h.secretsCache, config.Spec)
+	credentials, err := aks.GetSecrets(h.secretsCache, &config.Spec)
 	if err != nil {
 		return config, err
 	}
-	resourceGroupsClient, err := newResourceGroupsClient(azureAuthorizer, config.Spec)
-	if err != nil {
-		return config, err
-	}
-	resourceClusterClient, err := newClustersClient(azureAuthorizer, config.Spec)
+
+	resourceClusterClient, err := aks.NewClusterClient(credentials)
 	if err != nil {
 		return config, err
 	}
 
-	if existsCluster(ctx, resourceClusterClient, config.Spec) {
-		if err := removeCluster(ctx, resourceClusterClient, config.Spec); err != nil {
+	if aks.ExistsCluster(ctx, resourceClusterClient, &config.Spec) {
+		if err = aks.RemoveCluster(ctx, resourceClusterClient, &config.Spec); err != nil {
 			return config, fmt.Errorf("error removing cluster [%s] message %v", config.Spec.ClusterName, err)
 		}
 	}
 
-	logrus.Infof("Removing resource group [%s] for cluster [%s]", config.Spec.ResourceGroup, config.Spec.ClusterName)
+	logrus.Infof("Cluster [%s] was removed successfully", config.Spec.ClusterName)
 
-	if existsResourceGroup(ctx, resourceGroupsClient, config.Spec.ResourceGroup) {
```

Comment:
@cmurphy as we discussed, `Resource Group` will be kept after removing cluster

Reply:
Since this refactor is bundling in this change, could you make a note of this in the commit message?

Reply:
Added to commit message